### ISSUE TITLE
Azcopy error CannotVerifyCopySource

### ISSUE THREAD

**Issue by swapnilrkambli** [Azure/azure-storage-azcopy#1425]
### Which version of the AzCopy was used? 
AzCopy/10.9.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run?
& c:\tools\azcopy copy $sourceSASURI $destinationSASURI --s2s-preserve-access-tier --overwrite=ifSourceNewer --recursive

### What problem was encountered?
 Code: CannotVerifyCopySource

Source(East US2) and destination(West Europe) storage accounts have firewall enabled. Machine where Azcopy is being run  is added into both storage account firewalls. Azcopy logs contains Public IP address for the unauthorized access, Same was given by MS support to be added into storage account firewall rules. once the IP rule is added the azcopy runs without issue.

The issue is requirement for this firewall rule addition is not documented and there is no programmatic way to find out the IP address that needs to be added.

### Have you found a mitigation/solution?
After raising issue to MS support, Support gave us the destination storage account IP addresses to be added into source storage account firewall rule. once the rule is added AzCopy runs without any issue.

**Comment by adreed-msft** [comment]
tl;dr: Scrape the logging for the IP of the destination and give the user a warning when we think that auth is being denied due to a firewall
